### DISCUSSION TITLE
Few improvements for user experiences

### DISCUSSION THREAD

**Discussion by ghostersk** [celzero/rethink-app#791] [Ideas]
Hello, I love your app, it makes me feel much more in control over my android phone against all the spying.
I have few suggestions what would make this even greater app, worth to put it to Pro sale in app store for £10+

1. I can see already you are working on Whitelist in the app? As there is some option in settings with note it is comming...
   - It would be great if I can whitelist URLs rather than canceling whole blocklist where it gets blocked.
2. It would be awesome to get some quick menu shortcut for apps on desktop and app menu where if i hold it for couple seconds i would have option to trigger ON/OFF app Bypass for this RethninkApp filter, instead of me needing to going always to ap and search for app I want to block/unblock temporary when I block all apps by default.
3. I am not sure if this is possible, but maybe. When URL is blocked, instead of sending it to 0.0.0.0 what I believe it is currently set to go, would it be possible to set custom URL, for example I could send it to my website so I know it was blocked by the RethinkApp.
- Or even better would be if instead of DNS not found page RethinkApp could return page that it was blocked be Rethink and what Filter blocked it or reason for it to be blocked what would make this much better user experience.
   This could also give option to whitelist the URL in app?
4. I am not sure if it is because it is run in VPN tunel but for some reason I need to always disabled it completely to make my Hotspot work, once I connect I can use the RethinkDNS again. It seems block DHCP. 
 - It would be great if firewall can allow traffic on local IP addresses or allow DHCP, DNS and Time for LAN
 - 
Thank you. Great Job guys.

**Comment by ignoramous** [comment]
Hi there,

Thanks for your kind words.

1. [`v054` released 2 weeks ago](https://rethinkdns.com/download) with per-app domain allowlist/denylist feature. Check it out and let me know what you think (:
2. Shortcuts are planned; #402 
3. Won't work for HTTPS / encrypted connections without installing a custom certificate (generated by Rethink). This is a security issue, but doable. No plans as of yet as it is a major security hole for what's a minor feature, really.
4. Not sure how Rethink blocks Hotspot. Android OS guarantees that they don't mix. May be you're on a custom ROM that has broken this?
5. Local DNS queries (`.lan`, `.local`, `.internal`) are already split-tunneled through to DHCP-provisioned resolver in `v054`, but proper LAN access for apps is coming, too (apps connecting to private IPs will be allowed to *bypass* Rethink's tunnel, because that's the only way for apps to discover and talk to local endpoints, afaik): #26
